errors/errors.go

```diff
@@ -2,42 +2,41 @@ package errors
 
 import (
 	"fmt"
-	"strings"
-
 	"github.com/pkg/errors"
 )
 
 var (
 	// InternalErr represents a general case issue that cannot be
 	// categorized as any of the below cases.
-	InternalErr = RegisterNew(0, "internal")
+	// We start as 1 as 0 is reserved for non-errors
+	InternalErr = Register(1, "internal")
```

Comment:
Why did you change the internal error to 1?

The design of errors was relying on the internal error code being a zero value. This is important for the non weave errors, because they do not explicitly provide an error code.

Reply:
This is a good question.
My only hard requirements here is that the abci interface (deliver/checktx) *must* return code 0 on success and code > 0 on failure, as Roman referred to in the comment.

That said, I will make a suggestion below

Reply:
The error is returned here, as well as identical code for CheckTx:


```go
func DeliverTxError(err error, debug bool) abci.ResponseDeliverTx {
	tm := errors.Wrap(err)
	log := tm.ABCILog()
	if debug {
		log = fmt.Sprintf("%v", tm)
	}

	return abci.ResponseDeliverTx{
		Code: tm.ABCICode(),
		Log:  log,
	}
}
```

And I think Piotr's three points above are the three ways to override.
I think (1) which would change the `Code: tm.ABCICode(),` line is a bit more fragile, as we much make sure to update in all (two) places... 

I would fix it in the Wrap function. Also, note that this has no description and will delegate to the `deprecatedLegacyWrap`, which does already use 1 as a default for internal errors.

Updating `wrappedError.ABCICode` is as easy as updating the value for internal errors as Roman did. I think this is fine in all cases as is.

```go
func (e *wrappedError) ABCICode() uint32 {
	if e.Parent == nil {
		return InternalErr.code
	}
	type coder interface {
		ABCICode() uint32
	}
	if p, ok := e.Parent.(coder); ok {
		return p.ABCICode()
	}
	return InternalErr.code
}
```

So.... in conclusion, I think Roman's approach of changing the error code is proper, but we clearly need:
*  a test case of handling both wrapped and stdlib errors in `DeliverTxError` (ensure code always proper)
* reconsider why we use `tm := errors.Wrap(err)` and thus `deprecatedLegacyWrap` in `DeliverTxError`